### DISCUSSION TITLE
how to "where" on relations

### DISCUSSION THREAD

**Discussion by ghost** [adonisjs/core#2854] [Help]
problem in line 5 "user.last_name"

await Contract.query()
      .preload("user")
      .orderBy("created_at")
      .where("title", "like", `%${params.searchPayload}%`)
      .orWhere("user.last_name", "like", `%${params.searchPayload}%`)
      .paginate(params.page, params.perPage);

![image](https://user-images.githubusercontent.com/58485839/123634215-69802380-d82f-11eb-9078-08a209cacd86.png)

**Comment by RomainLanz** [comment]
Hey @mahdiirar! 👋 

The `preload` method takes a callback to apply conditions.

```ts
const users = await User
  .query()
  .preload('profile', (profileQuery) => {
    profileQuery.where('isActive', true)
  })
```

📚 https://docs.adonisjs.com/guides/models/relationships#preload-relationship
